src/test/java/org/elasticsearch/index/query/SimpleQueryStringBuilderTest.java

```diff
@@ -0,0 +1,164 @@
+
+public class SimpleQueryStringBuilderTest extends BaseQueryTestCase<SimpleQueryStringBuilder> {
+
+    @Override
+    protected SimpleQueryStringBuilder createTestQueryBuilder() {
+        SimpleQueryStringBuilder qb = new SimpleQueryStringBuilder(randomAsciiOfLength(8));
+        if (randomBoolean()) qb.queryName(randomAsciiOfLength(8));
+        if (randomBoolean()) qb.analyzeWildcard(randomBoolean());
+        if (randomBoolean()) qb.lenient(randomBoolean());
+        if (randomBoolean()) qb.lowercaseExpandedTerms(randomBoolean());
+        if (randomBoolean()) qb.locale(randomLocale(getRandom()));
+        if (randomBoolean()) qb.minimumShouldMatch(randomFrom(new String[]{"1", "-1", "75%", "-25%", "2<75%", "2<-25%"}));
+        if (randomBoolean()) qb.analyzer("simple");
+        if (randomBoolean()) qb.defaultOperator(randomFrom(Operator.AND, Operator.OR));
+
+        if (randomBoolean()) {
+            Set<SimpleQueryStringFlag> flags = new HashSet<>();
+            int size = randomIntBetween(0, SimpleQueryStringFlag.values().length);
+            for (int i = 0; i < size; i++) {
+                randomFrom(SimpleQueryStringFlag.values());
+            }
+            qb.flags(flags.toArray(new SimpleQueryStringFlag[flags.size()]));
+        }
+
+        int fieldCount = randomIntBetween(0, 10);
+        for (int i = 0; i < fieldCount; i++) { 
+            if (randomBoolean()) {
+                qb.field(randomAsciiOfLength(8));
+            } else {
+                qb.field(randomAsciiOfLength(8), 2.0f / randomIntBetween(1, 20));
+            }
+        }
+        return qb;
+    }
+
+    // Check operator handling, and default field handling.
+    @Test
+    public void testDefaultOperatorHandling() {
+        SimpleQueryStringBuilder qb = new SimpleQueryStringBuilder("The quick brown fox.");
+        BooleanQuery boolQuery = (BooleanQuery) qb.toQuery(createContext());
+        assertThat(shouldClauses(boolQuery), is(4));
+        
+        qb.defaultOperator(Operator.AND);
+        boolQuery = (BooleanQuery) qb.toQuery(createContext());
+        assertThat(shouldClauses(boolQuery), is(0));
+        
+        qb.defaultOperator(Operator.OR);
+        boolQuery = (BooleanQuery) qb.toQuery(createContext());
+        assertThat(shouldClauses(boolQuery), is(4));
+    }
+        
+    /*
+     * This assumes that Lucene query parsing is being checked already, adding checks only for our parsing extensions.
+     *
+     * Also this relies on {@link SimpleQueryStringTests} to test most of the actual functionality of query parsing.
+     */
+    @Override
+    protected void assertLuceneQuery(SimpleQueryStringBuilder queryBuilder, Query query, QueryParseContext context) {
+        if (queryBuilder.queryName() != null) {
+            Query namedQuery = context.copyNamedFilters().get(queryBuilder.queryName());
+            assertThat(namedQuery, equalTo(query));
+        }
+
+        if (query instanceof BooleanQuery) {
```

Comment:
I think you need to rebase here, you need to implement `createExpectedQuery` which returns, based on the builder, the expected lucene query, then the equal comparison happens under the hood. `assertLuceneQuery` is more for additional query specific checks now.